influxdb/influxdb.go

```diff
@@ -78,24 +79,23 @@ func (ic *InfluxdbCollector) GetConfigPolicy() (*cpolicy.ConfigPolicy, error) {
 }
 
 // GetMetricTypes returns list of metrics based on influxDB system monitoring
-func (ic *InfluxdbCollector) GetMetricTypes(cfg plugin.PluginConfigType) ([]plugin.PluginMetricType, error) {
-	mts := []plugin.PluginMetricType{}
+func (ic *InfluxdbCollector) GetMetricTypes(cfg plugin.ConfigType) ([]plugin.MetricType, error) {
+	mts := []plugin.MetricType{}
 
 	ic.init(cfg)
 	ic.getStatistics()  // get statistical information about influxDB
 	ic.getDiagnostics() // get diagnostic information about influxDB
 
 	for ns, dat := range ic.data {
-		mts = append(mts, plugin.PluginMetricType{Namespace_: splitNamespace(ns), Tags_: dat.tags})
+		mts = append(mts, plugin.MetricType{Namespace_: core.NewNamespace(splitNamespace(ns)...), Tags_: dat.tags})
 	}
 
 	return mts, nil
 }
 
 // CollectMetrics collects given metrics
-func (ic *InfluxdbCollector) CollectMetrics(mts []plugin.PluginMetricType) ([]plugin.PluginMetricType, error) {
-	metrics := []plugin.PluginMetricType{}
-	hostname, _ := os.Hostname()
+func (ic *InfluxdbCollector) CollectMetrics(mts []plugin.MetricType) ([]plugin.MetricType, error) {
+	metrics := []plugin.MetricType{}
```

Comment:
Consider populating ```mts``` from ```CollectMetrics(mts []plugin.MetricType)``` parameter instead of creating new instance. This way description, units and tags will stay.